panic on SIGUSR1

I tried to find this to avoid a duplicate bug report, but didn't.

So here it is:

```
2014/03/26 21:07:49 Queue report initiated.
panic: interface conversion: interface is nil, not []interface {}

goroutine 40 [running]:
runtime.panic(0x848680, 0xc2112f6d80)
    /usr/lib64/go1.2/src/pkg/runtime/panic.c:266 +0xb6
github.com/mozilla-services/heka/pipeline.(*PipelineConfig).FormatTextReport(0xc210057400, 0x8f4690, 0xf, 0xc210067200, 0x1077, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/report.go:244 +0x1c3
github.com/mozilla-services/heka/pipeline.(*PipelineConfig).allReportsStdout(0xc210057400)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/report.go:226 +0x5d
created by github.com/mozilla-services/heka/pipeline.Run
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/pipeline_runner.go:246 +0x1f60

goroutine 1 [chan receive]:
github.com/mozilla-services/heka/pipeline.Run(0xc210057400)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/pipeline_runner.go:234 +0xe4b
main.main()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/cmd/hekad/main.go:160 +0x9a6

goroutine 3 [syscall]:
os/signal.loop()
    /usr/lib64/go1.2/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
    /usr/lib64/go1.2/src/pkg/os/signal/signal_unix.go:27 +0x31

goroutine 4 [finalizer wait]:
runtime.park(0x4144d0, 0x1009830, 0xff4308)
    /usr/lib64/go1.2/src/pkg/runtime/proc.c:1342 +0x66
runfinq()
    /usr/lib64/go1.2/src/pkg/runtime/mgc0.c:2276 +0x84
runtime.goexit()
    /usr/lib64/go1.2/src/pkg/runtime/proc.c:1394

goroutine 6 [select]:
github.com/mozilla-services/heka/plugins/tcp.(*TcpOutput).Run(0xc2100532c0, 0x7f9be0d71090, 0xc2100bd300, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:375 +0x614
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bd300, 0x7f9be0d720b0, 0xc210057400, 0xc2100460a0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:486 +0x646
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 7 [chan receive]:
github.com/mozilla-services/heka/plugins.(*LogOutput).Run(0xc21009ed80, 0x7f9be0d71090, 0xc2100bd180, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/log_output.go:46 +0x7c
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bd180, 0x7f9be0d720b0, 0xc210057400, 0xc2100460a0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:486 +0x646
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 8 [chan receive]:
github.com/mozilla-services/heka/plugins/statsd.(*StatFilter).Run(0xc21009a500, 0x7f9be0d71760, 0xc2100bda80, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/statsd/stat_filter.go:85 +0x137
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bda80, 0x7f9be0d720b0, 0xc210057400, 0xc210057470)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:483 +0x223
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 9 [chan receive]:
github.com/mozilla-services/heka/plugins/statsd.(*StatFilter).Run(0xc21009aa00, 0x7f9be0d71760, 0xc2100bdb00, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/statsd/stat_filter.go:85 +0x137
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bdb00, 0x7f9be0d720b0, 0xc210057400, 0xc210057470)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:483 +0x223
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 10 [chan receive]:
github.com/mozilla-services/heka/plugins/statsd.(*StatFilter).Run(0xc21009a000, 0x7f9be0d71760, 0xc2100bd100, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/statsd/stat_filter.go:85 +0x137
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bd100, 0x7f9be0d720b0, 0xc210057400, 0xc210057470)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:483 +0x223
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 11 [select]:
github.com/mozilla-services/heka/pipeline.(*CounterFilter).Run(0xc210096b40, 0x7f9be0d71760, 0xc2100bd280, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/counter_filter.go:66 +0x1bb
github.com/mozilla-services/heka/pipeline.(*foRunner).Starter(0xc2100bd280, 0x7f9be0d720b0, 0xc210057400, 0xc210057470)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:483 +0x223
created by github.com/mozilla-services/heka/pipeline.(*foRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:450 +0x95

goroutine 12 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 13 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 14 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 15 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 16 [sleep]:
time.Sleep(0x1dcd6500)
    /usr/lib64/go1.2/src/pkg/runtime/time.goc:31 +0x31
github.com/mozilla-services/heka/plugins/tcp.(*TcpOutput).StreamOutput(0xc2100532c0, 0xc2100534d0, 0xc2100c3060, 0xc2100c30c0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:313 +0x40c
created by github.com/mozilla-services/heka/plugins/tcp.(*TcpOutput).Run
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:372 +0x1b7

goroutine 17 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 18 [syscall]:
runtime.goexit()
    /usr/lib64/go1.2/src/pkg/runtime/proc.c:1394

goroutine 19 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·018()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:258 +0xaa
created by github.com/mozilla-services/heka/pipeline.(*MatchRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:294 +0x8b

goroutine 20 [chan receive]:
github.com/mozilla-services/heka/pipeline.(*DiagnosticTracker).Run(0xc210064990)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/packet_tracking.go:105 +0xc9
created by github.com/mozilla-services/heka/pipeline.Run
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/pipeline_runner.go:215 +0x9ce

goroutine 21 [chan receive]:
github.com/mozilla-services/heka/pipeline.(*DiagnosticTracker).Run(0xc2100649c0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/packet_tracking.go:105 +0xc9
created by github.com/mozilla-services/heka/pipeline.Run
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/pipeline_runner.go:216 +0x9e8

goroutine 22 [select]:
github.com/mozilla-services/heka/pipeline.func·016()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:100 +0x892
created by github.com/mozilla-services/heka/pipeline.(*messageRouter).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/router.go:174 +0x63

goroutine 23 [IO wait]:
net.runtime_pollWait(0x7f9be0d709d8, 0x72, 0x0)
    /usr/lib64/go1.2/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc21005e0d0, 0x72, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc21005e0d0, 0xb, 0x7f9be0d6e110)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).accept(0xc21005e070, 0x9b0f60, 0x0, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_unix.go:382 +0x2c2
net.(*TCPListener).AcceptTCP(0xc210000430, 0x0, 0x40cc41, 0x77d7c0)
    /usr/lib64/go1.2/src/pkg/net/tcpsock_posix.go:233 +0x47
net.(*TCPListener).Accept(0xc210000430, 0xc2100bb581, 0x8, 0x4138da, 0x83fc20)
    /usr/lib64/go1.2/src/pkg/net/tcpsock_posix.go:243 +0x27
github.com/mozilla-services/heka/plugins/tcp.(*TcpInput).Run(0xc21005eaf0, 0x7f9be0d70ab0, 0xc21005e230, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_input.go:204 +0xd2
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc21005e230, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 24 [IO wait]:
net.runtime_pollWait(0x7f9be0d70930, 0x72, 0x0)
    /usr/lib64/go1.2/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc21005e4c0, 0x72, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc21005e4c0, 0xb, 0x7f9be0d6e110)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).Read(0xc21005e460, 0xc2100c5000, 0x10000, 0x10000, 0x0, ...)
    /usr/lib64/go1.2/src/pkg/net/fd_unix.go:204 +0x2a0
net.(*conn).Read(0xc210000580, 0xc2100c5000, 0x10000, 0x10000, 0x1e0d80f00, ...)
    /usr/lib64/go1.2/src/pkg/net/net.go:122 +0xc5
github.com/mozilla-services/heka/pipeline.(*streamParserBuffer).read(0xc2100c0740, 0x7f9be0d80f00, 0xc210000580, 0xc2101146c0, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:105 +0x1e9
github.com/mozilla-services/heka/pipeline.(*MessageProtoParser).Parse(0xc2100c29f0, 0x7f9be0d80f00, 0xc210000580, 0x7f9be0d80f00, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:265 +0x8f
github.com/mozilla-services/heka/pipeline.NetworkMessageProtoParser(0x7f9be0d70d28, 0xc210000580, 0x7f9be0d70dd8, 0xc2100c29f0, 0x7f9be0d70ab0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/net_utils.go:103 +0x9f
github.com/mozilla-services/heka/plugins/udp.(*UdpInput).Run(0xc21009dbe0, 0x7f9be0d70ab0, 0xc21005e540, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/udp/udp_input.go:138 +0x382
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc21005e540, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 25 [IO wait]:
net.runtime_pollWait(0x7f9be0d70888, 0x72, 0x0)
    /usr/lib64/go1.2/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc2100458b0, 0x72, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc2100458b0, 0xb, 0x7f9be0d6e110)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).Read(0xc210045850, 0xc2100d5000, 0x10000, 0x10000, 0x0, ...)
    /usr/lib64/go1.2/src/pkg/net/fd_unix.go:204 +0x2a0
net.(*conn).Read(0xc21009e078, 0xc2100d5000, 0x10000, 0x10000, 0x414398, ...)
    /usr/lib64/go1.2/src/pkg/net/net.go:122 +0xc5
github.com/mozilla-services/heka/pipeline.(*streamParserBuffer).read(0xc2100acf40, 0x7f9be0d80f00, 0xc21009e078, 0x7f9bdc43de90, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:105 +0x1e9
github.com/mozilla-services/heka/pipeline.(*TokenParser).Parse(0xc210042a00, 0x7f9be0d80f00, 0xc21009e078, 0x7f9be0d80f00, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:124 +0x8f
github.com/mozilla-services/heka/pipeline.NetworkPayloadParser(0x7f9be0d70d28, 0xc21009e078, 0x7f9be0d714d0, 0xc210042a00, 0x7f9be0d70ab0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/net_utils.go:63 +0x96
github.com/mozilla-services/heka/plugins/udp.(*UdpInput).Run(0xc210098aa0, 0x7f9be0d70ab0, 0xc210045930, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/udp/udp_input.go:138 +0x382
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc210045930, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 26 [IO wait]:
net.runtime_pollWait(0x7f9be0d707e0, 0x72, 0x0)
    /usr/lib64/go1.2/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc2100a28b0, 0x72, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc2100a28b0, 0xb, 0x7f9be0d6e110)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).Read(0xc2100a2850, 0xc2100ed000, 0x10000, 0x10000, 0x0, ...)
    /usr/lib64/go1.2/src/pkg/net/fd_unix.go:204 +0x2a0
net.(*conn).Read(0xc21009e840, 0xc2100ed000, 0x10000, 0x10000, 0x414398, ...)
    /usr/lib64/go1.2/src/pkg/net/net.go:122 +0xc5
github.com/mozilla-services/heka/pipeline.(*streamParserBuffer).read(0xc2100a1f80, 0x7f9be0d80f00, 0xc21009e840, 0xc2115c2960, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:105 +0x1e9
github.com/mozilla-services/heka/pipeline.(*TokenParser).Parse(0xc2100e51e0, 0x7f9be0d80f00, 0xc21009e840, 0x7f9be0d80f00, 0x0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stream_parser.go:124 +0x8f
github.com/mozilla-services/heka/pipeline.NetworkPayloadParser(0x7f9be0d70d28, 0xc21009e840, 0x7f9be0d714d0, 0xc2100e51e0, 0x7f9be0d70ab0, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/net_utils.go:63 +0x96
github.com/mozilla-services/heka/plugins/udp.(*UdpInput).Run(0xc2100ab1e0, 0x7f9be0d70ab0, 0xc2100a2930, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/udp/udp_input.go:138 +0x382
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc2100a2930, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 27 [IO wait]:
net.runtime_pollWait(0x7f9be0d70738, 0x72, 0x0)
    /usr/lib64/go1.2/src/pkg/runtime/netpoll.goc:116 +0x6a
net.(*pollDesc).Wait(0xc2100a2b50, 0x72, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:81 +0x34
net.(*pollDesc).WaitRead(0xc2100a2b50, 0xb, 0x7f9be0d6e110)
    /usr/lib64/go1.2/src/pkg/net/fd_poll_runtime.go:86 +0x30
net.(*netFD).accept(0xc2100a2af0, 0x9b0f60, 0x0, 0x7f9be0d6e110, 0xb)
    /usr/lib64/go1.2/src/pkg/net/fd_unix.go:382 +0x2c2
net.(*TCPListener).AcceptTCP(0xc21009e9c8, 0x0, 0x40cc41, 0x77d7c0)
    /usr/lib64/go1.2/src/pkg/net/tcpsock_posix.go:233 +0x47
net.(*TCPListener).Accept(0xc21009e9c8, 0xc21007fcf7, 0xf, 0x4138da, 0x83fc20)
    /usr/lib64/go1.2/src/pkg/net/tcpsock_posix.go:243 +0x27
github.com/mozilla-services/heka/plugins/tcp.(*TcpInput).Run(0xc2100a2a10, 0x7f9be0d70ab0, 0xc2100a2bd0, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_input.go:204 +0xd2
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc2100a2bd0, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 28 [select]:
github.com/mozilla-services/heka/pipeline.(*StatAccumInput).Run(0xc210096c30, 0x7f9be0d70ab0, 0xc2100455b0, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/stat_accum_input.go:142 +0x487
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc2100455b0, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 29 [select]:
github.com/mozilla-services/heka/plugins/logstreamer.(*LogstreamerInput).Run(0xc21007ebe0, 0x7f9be0d70ab0, 0xc2100aaa10, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/logstreamer/logstreamer_input.go:215 +0xd74
github.com/mozilla-services/heka/pipeline.(*iRunner).Starter(0xc2100aaa10, 0x7f9be0d720b0, 0xc210057400, 0xc2100574e0)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:189 +0x19c
created by github.com/mozilla-services/heka/pipeline.(*iRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:170 +0xb7

goroutine 30 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·010()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:312 +0x111
created by github.com/mozilla-services/heka/pipeline.(*dRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:330 +0x108

goroutine 31 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·010()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:312 +0x111
created by github.com/mozilla-services/heka/pipeline.(*dRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:330 +0x108

goroutine 32 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·010()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:312 +0x111
created by github.com/mozilla-services/heka/pipeline.(*dRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:330 +0x108

goroutine 33 [select]:
github.com/mozilla-services/heka/plugins/logstreamer.(*LogstreamInput).Run(0xc2100c37e0, 0x7f9be0d70ab0, 0xc2100aaa10, 0x7f9be0d720b0, 0xc210057400, ...)
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/logstreamer/logstreamer_input.go:340 +0x318
created by github.com/mozilla-services/heka/plugins/logstreamer.(*LogstreamerInput).Run
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/plugins/logstreamer/logstreamer_input.go:207 +0x3f4

goroutine 34 [chan receive]:
github.com/mozilla-services/heka/pipeline.func·010()
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:312 +0x111
created by github.com/mozilla-services/heka/pipeline.(*dRunner).Start
    /home/david/src/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/plugin_runners.go:330 +0x108
```

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -4,6 +4,9 @@
 Bug Handling
 ------------
 
+* Fixed SIGUSR1 triggered text report to stdout to work with updated
+  heka.all-report JSON data structure (issue #762).
+
 * MultiDecoder now uses message.GetType() instead of Message.Type so
   emitted message type will be generated correctly (issue #761).
 

--- a/pipeline/report.go
+++ b/pipeline/report.go
@@ -241,21 +241,25 @@ func (pc *PipelineConfig) FormatTextReport(report_type, payload string) string {
 	json.Unmarshal([]byte(payload), &m)
 
 	fullReport := make([]string, 0)
-	for _, row := range m["reports"].([]interface{}) {
-		pluginReport := make([]string, 0)
-		pluginReport = append(pluginReport,
-			fmt.Sprintf("%s:", (row.(map[string]interface{}))["Plugin"].(string)))
-		for _, colname := range header {
-			data := row.(map[string]interface{})[colname]
-			if data != nil {
-				pluginReport = append(pluginReport,
-					fmt.Sprintf("    %s: %s",
-						colname,
-						data.(map[string]interface{})["value"]))
+	categories := []string{"globals", "inputs", "decoders", "filters", "outputs"}
+	for _, cat := range categories {
+		fullReport = append(fullReport, fmt.Sprintf("\n====%s====", strings.Title(cat)))
+		for _, row := range m[cat].([]interface{}) {
+			pluginReport := make([]string, 0)
+			pluginReport = append(pluginReport,
+				fmt.Sprintf("%s:", (row.(map[string]interface{}))["Name"].(string)))
+			for _, colname := range header {
+				data := row.(map[string]interface{})[colname]
+				if data != nil {
+					pluginReport = append(pluginReport,
+						fmt.Sprintf("    %s: %s",
+							colname,
+							data.(map[string]interface{})["value"]))
+				}
 			}
-		}
 
-		fullReport = append(fullReport, strings.Join(pluginReport, "\n"))
+			fullReport = append(fullReport, strings.Join(pluginReport, "\n"))
+		}
 	}
 
 	stdout_report := fmt.Sprintf("========[%s]========\n%s\n========\n",

--- a/plugins/config_test.go
+++ b/plugins/config_test.go
@@ -172,11 +172,13 @@ func LoadFromConfigSpec(c gs.Context) {
 			err := pipeConfig.LoadFromConfigFile("./testsupport/config_test_defaults2.toml")
 			c.Expect(err, gs.IsNil)
 
-			data := `{"reports":[{"Plugin":"inputRecycleChan","InChanCapacity":{"value":"100", "representation":"count"},"InChanLength":{"value":"99", "representation":"count"}},{"Plugin":"injectRecycleChan","InChanCapacity":{"value":"100", "representation":"count"},"InChanLength":{"value":"98", "representation":"count"}},{"Plugin":"Router","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"ProcessMessageCount":{"value":"26", "representation":"count"}},{"Plugin":"ProtobufDecoder-0","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"}},{"Plugin":"ProtobufDecoder-1","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"}},{"Plugin":"ProtobufDecoder-2","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"}},{"Plugin":"ProtobufDecoder-3","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"}},{"Plugin":"DecoderPool-ProtobufDecoder","InChanCapacity":{"value":"4", "representation":"count"},"InChanLength":{"value":"4", "representation":"count"}},{"Plugin":"OpsSandboxManager","RunningFilters":{"value":"0", "representation":"count"},"ProcessMessageCount":{"value":"0", "representation":"count"},"InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"0", "representation":"ns"}},{"Plugin":"hekabench_counter","Memory":{"value":"20644", "representation":"B"},"MaxMemory":{"value":"20644", "representation":"B"},"MaxInstructions":{"value":"18", "representation":"count"},"MaxOutput":{"value":"0", "representation":"B"},"ProcessMessageCount":{"value":"0", "representation":"count"},"InjectMessageCount":{"value":"0", "representation":"count"},"ProcessMessageAvgDuration":{"value":"0", "representation":"ns"},"TimerEventAvgDuration":{"value":"78532", "representation":"ns"},"InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"445", "representation":"ns"}},{"Plugin":"LogOutput","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"406", "representation":"ns"}},{"Plugin":"DashboardOutput","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"336", "representation":"ns"}}]} `
+			data := `{"globals":[{"Name":"inputRecycleChan","InChanCapacity":{"value":"100", "representation":"count"},"InChanLength":{"value":"99", "representation":"count"}},{"Name":"injectRecycleChan","InChanCapacity":{"value":"100", "representation":"count"},"InChanLength":{"value":"98", "representation":"count"}},{"Name":"Router","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"ProcessMessageCount":{"value":"26", "representation":"count"}}], "inputs": [{"Name": "TcpInput"}], "decoders": [{"Name":"ProtobufDecoder","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"}}], "filters": [{"Name":"OpsSandboxManager","RunningFilters":{"value":"0", "representation":"count"},"ProcessMessageCount":{"value":"0", "representation":"count"},"InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"0", "representation":"ns"}},{"Name":"hekabench_counter","Memory":{"value":"20644", "representation":"B"},"MaxMemory":{"value":"20644", "representation":"B"},"MaxInstructions":{"value":"18", "representation":"count"},"MaxOutput":{"value":"0", "representation":"B"},"ProcessMessageCount":{"value":"0", "representation":"count"},"InjectMessageCount":{"value":"0", "representation":"count"},"ProcessMessageAvgDuration":{"value":"0", "representation":"ns"},"TimerEventAvgDuration":{"value":"78532", "representation":"ns"},"InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"445", "representation":"ns"}}], "outputs": [{"Name":"LogOutput","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"406", "representation":"ns"}},{"Name":"DashboardOutput","InChanCapacity":{"value":"50", "representation":"count"},"InChanLength":{"value":"0", "representation":"count"},"MatchChanCapacity":{"value":"50", "representation":"count"},"MatchChanLength":{"value":"0", "representation":"count"},"MatchAvgDuration":{"value":"336", "representation":"ns"}}]} `
 
 			report := pipeConfig.FormatTextReport("heka.all-report", data)
 
 			expected := `========[heka.all-report]========
+
+====Globals====
 inputRecycleChan:
     InChanCapacity: 100
     InChanLength: 99
@@ -187,21 +189,16 @@ Router:
     InChanCapacity: 50
     InChanLength: 0
     ProcessMessageCount: 26
-ProtobufDecoder-0:
+
+====Inputs====
+TcpInput:
+
+====Decoders====
+ProtobufDecoder:
     InChanCapacity: 50
     InChanLength: 0
-ProtobufDecoder-1:
-    InChanCapacity: 50
-    InChanLength: 0
-ProtobufDecoder-2:
-    InChanCapacity: 50
-    InChanLength: 0
-ProtobufDecoder-3:
-    InChanCapacity: 50
-    InChanLength: 0
-DecoderPool-ProtobufDecoder:
-    InChanCapacity: 4
-    InChanLength: 4
+
+====Filters====
 OpsSandboxManager:
     InChanCapacity: 50
     InChanLength: 0
@@ -223,6 +220,8 @@ hekabench_counter:
     MaxOutput: 0
     ProcessMessageAvgDuration: 0
     TimerEventAvgDuration: 78532
+
+====Outputs====
 LogOutput:
     InChanCapacity: 50
     InChanLength: 0

